lib/variable/include/scipp/variable/element_array_variable.tcc

```diff
@@ -108,6 +108,18 @@ bool ElementArrayModel<T>::equals(const Variable &a, const Variable &b) const {
           equals_impl(a.variances<T>(), b.variances<T>()));
 }
 
+/// Helper for implementing Variable::operator==.
+///
+/// This method is using virtual dispatch as a trick to obtain T, such that
+/// values<T> and variances<T> can be compared.
```

Comment:
Does it? Looks like the caller has to find `T`.

Reply:
That happens before the function is even called.
A bit nit picky and I am therefore resolving this conversation.